integration_tests/src/main/python/data_gen.py

```diff
@@ -684,6 +684,8 @@ def to_cast_string(spark_type):
         return 'TIMESTAMP'
     elif isinstance(spark_type, StringType):
         return 'STRING'
+    elif isinstance(spark_type, DecimalType):
+        return 'DECIMAL'
```

Comment:
Shouldn't this include the precision and scale in the cast type?

Reply:
yes, I should've noticed that we are using `cast` not `decimal` function. made the change